test/e2e/operator/nmstate_install_test.go

```diff
@@ -12,103 +12,87 @@ import (
-				Expect(err).ToNot(HaveOccurred(), "NMState CR with incorrect name is removed without error")
-			})
-			It("should ignore it", func() {
-				ds, err := daemonset.GetList(handlerLabels)
-				Expect(err).ToNot(HaveOccurred(), "Daemon set list is retreieved without error")
-				Expect(ds.Items).To(HaveLen(1), "and still only has one item")
+			It("should remove NMState with different name", func() {
+				Eventually(func() error {
+					return testenv.Client.Get(context.TODO(), types.NamespacedName{Name: differentNMState.Name}, &differentNMState)
+				}, 120*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())))
 			})
+
 		})
 		Context("and uninstalled", func() {
 			BeforeEach(func() {
-				uninstallNMState(defaultNMState)
+				uninstallNMState(defaultOperator.nmstate)
 			})
 			It("should uninstall handler and webhook", func() {
-				Eventually(func() bool {
-					_, err := daemonset.Get(handlerKey)
-					return apierrors.IsNotFound(err)
-				}, 120*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprint("Daemon Set for NMState handler should be removed, but is not"))
-				Eventually(func() bool {
-					_, err := deployment.Get(webhookKey)
-					return apierrors.IsNotFound(err)
-				}, 120*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprint("Deployment for NMState webhook should be removed, but is not"))
+				eventuallyOperandIsNotFound(defaultOperator)
 			})
 		})
-	})
-	Context("when another handler is installed with different namespace", func() {
-		var (
-			operatorNamespace = "nmstate-alt"
-		)
-		BeforeEach(func() {
-			installNMState(defaultNMState)
-			daemonset.GetEventually(handlerKey).Should(daemonset.BeReady())
-			installOperator(operatorNamespace)
-		})
-		AfterEach(func() {
-			uninstallNMState(defaultNMState)
-			uninstallOperator(operatorNamespace)
-			installOperator("nmstate")
-		})
-		It("should wait on the old one to be deleted", func() {
-			newDaemonSetKey := types.NamespacedName{Namespace: operatorNamespace, Name: "nmstate-handler"}
-			Eventually(func() error {
-				daemonSet := appsv1.DaemonSet{}
-				err := testenv.Client.Get(context.TODO(), newDaemonSetKey, &daemonSet)
-				return err
-			}, 180*time.Second, 1*time.Second).Should(SatisfyAny(Succeed(), WithTransform(apierrors.IsNotFound, BeFalse())))
+		Context("and another handler is installed with different namespace", func() {
+			var (
+				altOperator = newOperatorTestData("nmstate-alt")
+			)
+			BeforeEach(func() {
+				By("Wait for operand to be ready")
+				eventuallyOperandIsReady(defaultOperator)
+
+				By("Install other operator at alternative namespace")
+				installOperator(altOperator)
+			})
+			AfterEach(func() {
+				uninstallOperator(altOperator)
+				eventuallyOperandIsNotFound(altOperator)
+				uninstallNMStateAndWaitForDeletion(defaultOperator)
+				installOperator(defaultOperator)
+			})
+			It("should wait on the old one to be deleted", func() {
```

Comment:
Nit: I know this is not introduced by this PR, but IMO this title is not as descriptive as it could be.
`should wait for defaultOperator to be deleted before deploying new altOperator`  would be better.
Not a must for this PR of course

Reply:
Done